app/stac_api/serializers.py

```diff
@@ -0,0 +1,188 @@
+
+    def update(self, instance, validated_data):
+        """
+        Update and return an existing `Keyword` instance, given the validated data.
+        """
+
+        instance.name = validated_data.get('name', instance.name)
+
+        instance.save()
+        return instance
+
+
+class LinkSerializer(serializers.Serializer):
+
+    href = serializers.URLField()  # string
+    rel = serializers.CharField(max_length=30)  # string
+    link_type = serializers.CharField(allow_blank=True, max_length=150)  # string
+    title = serializers.CharField(allow_blank=True, max_length=255)  # string
+
+    class Meta:
+        model = Link
+        fields = ['href', 'rel', 'link_type', 'title']
+        # most likely not all fields necessary here, can be adapted
+
+    def create(self, validated_data):
+        """
+        Create and return a new `Link` instance, given the validated data.
+        """
+        return Collection.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        """
+        Update and return an existing `Link` instance, given the validated data.
+        """
+
+        instance.href = validated_data.get('href', instance.href)
+        instance.rel = validated_data.get('rel', instance.rel)
+        instance.link_type = validated_data.get('link_type', instance.link_type)
+        instance.title = validated_data.get('title', instance.title)
+
+        instance.save()
+        return instance
+
+
+class ProviderSerializer(serializers.Serializer):
+
+    name = serializers.CharField(allow_blank=False, max_length=200)  # string
+    description = serializers.CharField()  # string
+    roles = serializers.ListField(child=serializers.CharField(max_length=9))  # [string]
+    url = serializers.URLField()  # string
+
+    class Meta:
+        model = Provider
+        fields = ['name', 'description', 'roles', 'url']
+        # most likely not all fields necessary here, can be adapted
+
+    def create(self, validated_data):
+        """
+        Create and return a new `Provider` instance, given the validated data.
+        """
+        return Collection.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        """
+        Update and return an existing `Provider` instance, given the validated data.
+        """
+
+        instance.name = validated_data.get('name', instance.name)
+        instance.description = validated_data.get('description', instance.description)
+        instance.roles = validated_data.get('roles', instance.roles)
+        instance.url = validated_data.get('url', instance.url)
+
+        instance.save()
+        return instance
+
+
+class CollectionSerializer(serializers.Serializer):
+
+    # TODO: Do we have to take the automatically created primary_key field
+    # into account here, does it have to be serialized?
+    # (In 0001_initial.py, there is an auto-generated entry "serialize=False" for the
+    # primary-key field.)
```

Comment:
probably not (since it's not part or even allowed by the spec). the "public" unique field is the collection_name, we can make updates by filtering for that one

Reply:
OK, comment removed